### PR TITLE
Byte builtin

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
Test failing because `let x := 2` is not encoded correctly.

File: src/evm_builtins.rs [hunk 1]
```diff
@@ -60,7 +60,18 @@ impl Instructions for EVMInstructions {
...
+                let input = &arguments[1].name;
+                let shift_amount =
+                    format!("(bvsub #x{:064x} (bvmul #x{:064x} {byte_index}))", 248, 8);
+                single_return(format!(
+                    "(ite (bvugt {byte_index} #x{:064x}) #x{:064x} (bvand #x{:064x} (bvlshr {input} {shift_amount})))",
+                    31,
+                    0,
+                    0xff
+                ))
+            }
```
Comment (leonardoalt) [inline src/evm_builtins.rs:74]: lol the evm is quirky af why is byte 0 the MSB

**Review by leonardoalt — APPROVED** [review]

File: src/evm_builtins.rs (hunk 1 above)
Comment (chriseth) [reply src/evm_builtins.rs:74]: it's a byte array, you count from left to right - how else would you do it?

Comment (leonardoalt) [reply src/evm_builtins.rs:74]: 0 = LSB

**Pull request merged**